test/integration/consul-container/test/gateways/http_route_test.go

```diff
@@ -186,73 +185,229 @@ func TestHTTPRouteFlattening(t *testing.T) {
+		HTTPPort: 8080,
+		GRPCPort: 8081,
+	}, []string{
+		//customizes response code so we can distinguish between which service is responding
+		"-echo-debug-path", fooPath,
+		"-echo-server-default-params", fmt.Sprintf("status=%d", fooStatusCode),
+	})
+	barService := createService(t, cluster, &libservice.ServiceOpts{
+		Name: "bar",
+		ID:   "bar",
+		//TODO we can potentially get conflicts if these ports are the same
+		HTTPPort: 8079,
+		GRPCPort: 8078,
+	}, []string{
+		"-echo-debug-path", barPath,
+		"-echo-server-default-params", fmt.Sprintf("status=%d", barStatusCode),
+	},
+	)
+
+	namespace := getNamespace()
+	gatewayName := randomName("gw", 16)
+	invalidRouteName := randomName("route", 16)
+	validRouteName := randomName("route", 16)
+	fooUnrewritten := "/foo"
+	barUnrewritten := "/bar"
+
+	//write config entries
+	proxyDefaults := &api.ProxyConfigEntry{
+		Kind:      api.ProxyDefaults,
+		Name:      api.ProxyConfigGlobal,
+		Namespace: namespace,
+		Config: map[string]interface{}{
+			"protocol": "http",
+		},
+	}
+
+	_, _, err := client.ConfigEntries().Set(proxyDefaults, nil)
+	assert.NoError(t, err)
+
+	apiGateway := createGateway(gatewayName, "http", listenerPort)
+
+	fooRoute := &api.HTTPRouteConfigEntry{
+		Kind: api.HTTPRoute,
+		Name: invalidRouteName,
+		Parents: []api.ResourceReference{
+			{
+				Kind:      api.APIGateway,
+				Name:      gatewayName,
+				Namespace: namespace,
+			},
+		},
+		Hostnames: []string{
+			"test.foo",
+		},
+		Namespace: namespace,
+		Rules: []api.HTTPRouteRule{
+			{
+				Filters: api.HTTPFilters{
+					URLRewrite: &api.URLRewrite{
+						Path: fooPath,
+					},
+				},
+				Services: []api.HTTPService{
+					{
+						Name:      fooService.GetServiceName(),
+						Namespace: namespace,
+					},
+				},
+				Matches: []api.HTTPMatch{
+					{
+						Path: api.HTTPPathMatch{
+							Match: api.HTTPPathMatchPrefix,
+							Value: fooUnrewritten,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	barRoute := &api.HTTPRouteConfigEntry{
+		Kind: api.HTTPRoute,
+		Name: validRouteName,
+		Parents: []api.ResourceReference{
+			{
+				Kind:      api.APIGateway,
+				Name:      gatewayName,
+				Namespace: namespace,
+			},
+		},
+		Hostnames: []string{
+			"test.foo",
+		},
+		Namespace: namespace,
+		Rules: []api.HTTPRouteRule{
+			{
+				Filters: api.HTTPFilters{
+					URLRewrite: &api.URLRewrite{
+						Path: barPath,
+					},
+				},
+				Services: []api.HTTPService{
+					{
+						Name:      barService.GetServiceName(),
+						Namespace: namespace,
+					},
+				},
+				Matches: []api.HTTPMatch{
+					{
+						Path: api.HTTPPathMatch{
+							Match: api.HTTPPathMatchPrefix,
+							Value: barUnrewritten,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, _, err = client.ConfigEntries().Set(apiGateway, nil)
+	assert.NoError(t, err)
+	_, _, err = client.ConfigEntries().Set(fooRoute, nil)
+	assert.NoError(t, err)
+	_, _, err = client.ConfigEntries().Set(barRoute, nil)
+	assert.NoError(t, err)
```

Comment:
I'd use `require` here since the rest of the test is invalid if this assertion doesn't pass; otherwise, the rest of the test will continue running even if there was an error setting the config entry here